src/Interface/HasOrder.agda

```diff
@@ -0,0 +1,24 @@
+{-# OPTIONS --safe #-}
+
+module Interface.HasOrder where
+
+open import Level using (_⊔_ ; suc)
+open import Relation.Binary using (Rel)
+open import Relation.Binary.Structures using (IsPreorder ; IsPartialOrder)
+
+record HasPreorder {a ℓ ℓ₂} (A : Set a) (_≈_ : Rel A ℓ) : Set (a ⊔ ℓ ⊔ suc ℓ₂) where
+  infix 4 _≤_
+  field
+    _≤_ : Rel A ℓ₂
+    isPreorder : IsPreorder {A = A} _≈_ _≤_
```

Comment:
Preorder type class is included in case we want to use it for inequality on the `Epoch` type, which is not antisymmetric.